homeassistant/components/opentherm_gw/__init__.py

```diff
@@ -15,344 +24,251 @@
 
+
 _LOGGER = logging.getLogger(__name__)
 
 DOMAIN = 'opentherm_gw'
 
-ATTR_MODE = 'mode'
-ATTR_LEVEL = 'level'
-
-CONF_CLIMATE = 'climate'
-CONF_FLOOR_TEMP = 'floor_temperature'
-CONF_PRECISION = 'precision'
-
-DATA_DEVICE = 'device'
-DATA_GW_VARS = 'gw_vars'
-DATA_LATEST_STATUS = 'latest_status'
-DATA_OPENTHERM_GW = 'opentherm_gw'
-
-SIGNAL_OPENTHERM_GW_UPDATE = 'opentherm_gw_update'
-
-SERVICE_RESET_GATEWAY = 'reset_gateway'
-
-SERVICE_SET_CLOCK = 'set_clock'
-SERVICE_SET_CLOCK_SCHEMA = vol.Schema({
-    vol.Optional(ATTR_DATE, default=date.today()): cv.date,
-    vol.Optional(ATTR_TIME, default=datetime.now().time()): cv.time,
-})
-
-SERVICE_SET_CONTROL_SETPOINT = 'set_control_setpoint'
-SERVICE_SET_CONTROL_SETPOINT_SCHEMA = vol.Schema({
-    vol.Required(ATTR_TEMPERATURE): vol.All(vol.Coerce(float),
-                                            vol.Range(min=0, max=90)),
-})
-
-SERVICE_SET_GPIO_MODE = 'set_gpio_mode'
-SERVICE_SET_GPIO_MODE_SCHEMA = vol.Schema(vol.Any(
-    vol.Schema({
-        vol.Required(ATTR_ID): vol.Equal('A'),
-        vol.Required(ATTR_MODE): vol.All(vol.Coerce(int),
-                                         vol.Range(min=0, max=6)),
-    }),
-    vol.Schema({
-        vol.Required(ATTR_ID): vol.Equal('B'),
-        vol.Required(ATTR_MODE): vol.All(vol.Coerce(int),
-                                         vol.Range(min=0, max=7)),
-    }),
-))
-
-SERVICE_SET_LED_MODE = 'set_led_mode'
-SERVICE_SET_LED_MODE_SCHEMA = vol.Schema({
-    vol.Required(ATTR_ID): vol.In('ABCDEF'),
-    vol.Required(ATTR_MODE): vol.In('RXTBOFHWCEMP'),
-})
-
-SERVICE_SET_MAX_MOD = 'set_max_modulation'
-SERVICE_SET_MAX_MOD_SCHEMA = vol.Schema({
-    vol.Required(ATTR_LEVEL): vol.All(vol.Coerce(int),
-                                      vol.Range(min=-1, max=100))
-})
-
-SERVICE_SET_OAT = 'set_outside_temperature'
-SERVICE_SET_OAT_SCHEMA = vol.Schema({
-    vol.Required(ATTR_TEMPERATURE): vol.All(vol.Coerce(float),
-                                            vol.Range(min=-40, max=99)),
-})
-
-SERVICE_SET_SB_TEMP = 'set_setback_temperature'
-SERVICE_SET_SB_TEMP_SCHEMA = vol.Schema({
-    vol.Required(ATTR_TEMPERATURE): vol.All(vol.Coerce(float),
-                                            vol.Range(min=0, max=30)),
-})
-
 CLIMATE_SCHEMA = vol.Schema({
-    vol.Optional(CONF_NAME, default="OpenTherm Gateway"): cv.string,
     vol.Optional(CONF_PRECISION): vol.In([PRECISION_TENTHS, PRECISION_HALVES,
                                           PRECISION_WHOLE]),
     vol.Optional(CONF_FLOOR_TEMP, default=False): cv.boolean,
 })
 
 CONFIG_SCHEMA = vol.Schema({
-    DOMAIN: vol.Schema({
+    DOMAIN: cv.schema_with_slug_keys({
         vol.Required(CONF_DEVICE): cv.string,
         vol.Optional(CONF_CLIMATE, default={}): CLIMATE_SCHEMA,
-        vol.Optional(CONF_MONITORED_VARIABLES, default=[]): vol.All(
-            cv.ensure_list, [cv.string]),
+        vol.Optional(CONF_MONITORED_VARIABLES, default=[]): [vol.Any(
```

Comment:
There was a recent architecture decision that we should avoid user selectable monitored variables: https://github.com/home-assistant/architecture/blob/master/adr/0003-monitor-condition-and-data-selectors.md.

Can we motivate keeping this option here after reading the decision or should we remove the option?

Reply:
For the last few weeks/months, I haven't really kept up with HA development so I was not aware of this. All data is obtained from the device regardless of the configured `monitored_variables`, so the option is just there to reduce the amount of entities the integration exposes. As lovelace is now the preferred way to 'filter' this information, I will remove the option to select which data is exposed and create all entities by default.
home-assistant/home-assistant.io#9176 will be updated accordingly as well.